### PR TITLE
fix(pnpify): Load prettier with in-memory node_modules enabled

### LINKED ISSUE #1903: [Bug] Prettier is unable to load plugins in PNP environment
- [X] I'd be willing to implement a fix

**Describe the bug**

When using Prettier with plugins in a PNP environment, Prettier will not be able to find its plugins.

**To Reproduce**

We reproduced this on Discord with this repo:

https://github.com/oliversalzburg/prettier-plugin-pnp

Running `yarn prettier src/index.ts` will not remove the unused import, for which the `prettier-plugin-organize-imports` was installed.

**Additional context**

It was suggested that this is due to missing PNP functionality in the SDK wrapper. Enabling `usePnpif` in https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-pnpify/sources/generateSdk.ts#L119 should be a valid solution to the problem.

This should probably be enabled through a CLI flag in `yarn pnpify` to allow an adjusted SDK wrapper to be generated.

To actually fix the issue, I also need to supply the plugin search directory to Prettier manually. Otherwise it will search in a completely inapproriate location by default.

### PR DESCRIPTION
**Pull request by oliversalzburg**
**What's the problem this PR addresses?**
Using `--compat prettier,eslint`, the user can have the PnPify filesystem virtualization enabled in an SDK wrapper. This will allow users to get around packages (like `prettier`) that heavily rely on a `node_modules` file system layout to auto-load their plugins.

Fixes #1903

**How did you fix it?**
The template used for the wrapper, already has an argument to have the PnPify virtualization enabled. So far it appeares unused.

In the SDK generation path, I accept the same argument to have this virtualization enabled if requested from upstream.

I accept a new argument `--compat eslint,prettier,whatever` and when a base SDK name matches any of the given arguments, it will have the `usePnpify` option enabled.

**Checklist**

- [X] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-pnpify/sources/commands/SdkCommand.ts [hunk 1]
```diff
@@ -10,6 +10,9 @@ export default class SdkCommand extends Command {
   @Command.Rest()
   integrations: Array<string> = [];
 
+  @Command.Array(`--compat`, {description: `SDKs that should use PnPify`})
+  withPnpify: Array<string> = [];
```
Comment (larixer) [inline packages/yarnpkg-pnpify/sources/commands/SdkCommand.ts:14]: `withPnpify` should be renamed to `compat` here

File: packages/yarnpkg-pnpify/sources/commands/SdkCommand.ts [hunk 2]
```diff
@@ -113,6 +121,7 @@ export default class SdkCommand extends Command {
         onlyBase,
         configuration,
         verbose: this.verbose,
+        withPnpify: this.compat,
```
Comment (larixer) [inline packages/yarnpkg-pnpify/sources/commands/SdkCommand.ts:124]: @oliversalzburg should be `compat` as well, I think

**Review by larixer — CHANGES_REQUESTED** [review]

**Comment by arcanis** [comment]
I'm not a fan of making it an extra CLI argument that people have to use or not, but that we expect them to always set if they need to fix issues. At the same time, I'm not a fan of enabling PnPify either, as it's a bit intrusive on the fs layer ... could this be fixed on Prettier itself?

File: packages/yarnpkg-pnpify/sources/commands/SdkCommand.ts (hunk 2 above)
Comment (oliversalzburg) [reply packages/yarnpkg-pnpify/sources/commands/SdkCommand.ts:124]: How far should I pass it through as `compat`? In the final usage in the `TEMPLATE`, the name was `usePnpify`. At some point I will have to transition.

Comment (larixer) [reply packages/yarnpkg-pnpify/sources/commands/SdkCommand.ts:124]: @oliversalzburg "How far should I pass it through as `compat`". I think it should be "compat" everywhere, except in TEMPLATE it should be "usePnpify"

**Comment by oliversalzburg** [comment]
I'm sure it _could_ be fixed in Prettier, but it sounded quite unlikely to land in the forseeable future. 

They have an unmerged PR to fix issues with their current approach, where they explicitly mention not supporting PNP. @merceyz was active there and may have more insight into the underlying issue and the state of affairs.

For reference:
- https://github.com/prettier/prettier/issues/8474
- https://github.com/prettier/prettier/pull/8475

**Comment by merceyz** [comment]
> but it sounded quite unlikely to land in the forseeable future.

We could just make them query the pnpapi for all plugins until they make an actual fix, if they wont merge that the compat plugin is always an option

**Comment by larixer** [comment]
@arcanis 

> I'm not a fan of making it an extra CLI argument that people have to use or not, but that we expect them to always set if they need to fix issues. At the same time, I'm not a fan of enabling PnPify either, as it's a bit intrusive on the fs layer ... could this be fixed on Prettier itself?

I think these all are separate concerns. Improving prettier is one concern, but letting users to unblock themselves in certain situations with the help of in-memory `node_modules` is another concern. This PR is about the latter, not former.

**Review by larixer — APPROVED** [review]

**Pull request merged**